webapp/security/views.py

```diff
@@ -571,40 +571,67 @@ def cve(cve_id):
 
 # CVE API
 # ===
-def update_statuses(cve, data, packages, releases):
-    updated_statuses = []
+def update_statuses(cve, data, packages):
+    statuses = cve.packages
+
+    statuses_query = db_session.query(Status).filter(Status.cve_id == cve.id)
+    statuses_to_delete = {
+        f"{v.package_name}||{v.release_codename}": v
+        for v in statuses_query.all()
+    }
 
     for package_data in data.get("packages", []):
         name = package_data["name"]
-        package = packages.get(name) or Package(name=name)
-        package.source = package_data["source"]
-        package.ubuntu = package_data["ubuntu"]
-        package.debian = package_data["debian"]
-        packages[name] = package
 
-        statuses = cve.packages
+        if packages.get(name) is None:
+            package = Package(name=name)
+            package.source = package_data["source"]
+            package.ubuntu = package_data["ubuntu"]
+            package.debian = package_data["debian"]
+            packages[name] = package
+
+            db_session.add(package)
 
         for status_data in package_data["statuses"]:
+            update_status = False
             codename = status_data["release_codename"]
-            release = releases[codename]
 
-            status = statuses[name].get(codename) or Status(
-                cve=cve, package=package, release=release
-            )
+            status = statuses[name].get(codename)
+            if status is None:
+                update_status = True
+                status = Status(
+                    cve_id=cve.id, package_name=name, release_codename=codename
+                )
+            else:
+                del statuses_to_delete[f"{name}||{codename}"]
+
+            if status.status != status_data["status"]:
+                update_status = True
+                status.status = status_data["status"]
+
+            if status.description != status_data["description"]:
+                update_status = True
+                status.description = status_data["description"]
 
-            status.status = status_data["status"]
-            status.description = status_data["description"]
+            if "component" not in status_data:
+                status_data["component"] = None
+            if "pocket" not in status_data:
+                status_data["pocket"] = None
 
-            if "component" in status_data:
+            if status.component != status_data["component"]:
```

Comment:
You could remove the if statments above by doing `status_data.get("component")`, right?

Reply:
Done.